src/core/wrap.ts

```diff
@@ -0,0 +1,18 @@
+import { Metric } from "@clarity-types/data";
+import config from "@src/core/config";
+import * as metric from "@src/data/metric";
+
+// tslint:disable-next-line: ban-types
+export default function(method: Function): Function {
+    return function(): void {
+        let start = Date.now();
+        method.apply(this, arguments);
+        let duration = Date.now() - start;
+        metric.counter(Metric.Cost, duration);
```

Comment:
>counter [](start = 15, length = 7)

Doesn't seem like a counter to me. :)

Reply:
Please suggest a better name and I am happy to change the name :)

Reply:
Refactored per your suggestion in the last comment. Called it duration now.